diff --git a/Makefile b/Makefile
index 583bc25b..875775ba 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ ARCHFLAGS := $(ARCHFLAGS)
 
 OPTFLAGS  ?= -g -O
 
-CFLAGS    += -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
+CFLAGS    += -DWIN32_NO_DLL -DHAVE_STRUCT_TIMESPEC -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
 ifneq ($(findstring arm,$(shell uname -m)),)
 	CFLAGS += -DABC_MEMALIGN=4
 endif
@@ -142,7 +142,7 @@ ifneq ($(OS), FreeBSD)
 endif
 
 ifneq ($(findstring Darwin, $(shell uname)), Darwin)
-   LIBS += -lrt
+#   LIBS += -lrt
 endif
 
 ifdef ABC_USE_LIBSTDCXX
@@ -206,7 +206,12 @@ depend: $(DEP)
 
 clean:
 	@echo "$(MSG_PREFIX)\`\` Cleaning up..."
-	$(VERBOSE)rm -rvf $(PROG) lib$(PROG).a $(OBJ) $(GARBAGE) $(OBJ:.o=.d)
+	$(VERBOSE)rm -rvf $(OBJ:.o=.d)
+	$(VERBOSE)rm -rvf $(PROG)
+	$(VERBOSE)rm -rvf lib$(PROG).a
+	$(VERBOSE)rm -rvf $(OBJ)
+	$(VERBOSE)rm -rvf $(GARBAGE)
+
 
 tags:
 	etags `find . -type f -regex '.*\.\(c\|h\)'`
