diff --git a/Makefile b/Makefile
index ee4172f5..d1b64b3e 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ ARCHFLAGS := $(ARCHFLAGS)
 
 OPTFLAGS  ?= -g -O
 
-CFLAGS    += -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
+CFLAGS    += -DWIN32_NO_DLL -DHAVE_STRUCT_TIMESPEC -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
 ifneq ($(findstring arm,$(shell uname -m)),)
 	CFLAGS += -DABC_MEMALIGN=4
 endif
@@ -142,7 +142,7 @@ ifneq ($(OS), FreeBSD)
 endif
 
 ifneq ($(findstring Darwin, $(shell uname)), Darwin)
-   LIBS += -lrt
+#   LIBS += -lrt
 endif
 
 ifdef ABC_USE_LIBSTDCXX
