diff --git a/Makefile b/Makefile
index ee4172f5..272421c5 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ ARCHFLAGS := $(ARCHFLAGS)
 
 OPTFLAGS  ?= -g -O
 
-CFLAGS    += -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
+CFLAGS    += -DWIN32_NO_DLL -DHAVE_STRUCT_TIMESPEC -Wall -Wno-unused-function -Wno-write-strings -Wno-sign-compare $(ARCHFLAGS)
 ifneq ($(findstring arm,$(shell uname -m)),)
 	CFLAGS += -DABC_MEMALIGN=4
 endif
@@ -138,11 +138,11 @@ endif
 # LIBS := -ldl -lrt
 LIBS += -lm
 ifneq ($(OS), FreeBSD)
-  LIBS += -ldl
+#  LIBS += -ldl
 endif
 
 ifneq ($(findstring Darwin, $(shell uname)), Darwin)
-   LIBS += -lrt
+#   LIBS += -lrt
 endif
 
 ifdef ABC_USE_LIBSTDCXX
