diff --git a/passes/pmgen/ice40_dsp.cc b/passes/pmgen/ice40_dsp.cc
index 24134b1fb..a09243d55 100644
--- a/passes/pmgen/ice40_dsp.cc
+++ b/passes/pmgen/ice40_dsp.cc
@@ -73,11 +73,11 @@ void create_ice40_dsp(ice40_dsp_pm &pm)
 
 	// SB_MAC16 Input Interface
 	SigSpec A = st.sigA;
-	A.extend_u0(16, st.mul->getParam(ID::A_SIGNED).as_bool());
+	A.extend_u0(16, /*st.mul->getParam(ID::A_SIGNED).as_bool()*/ true );
 	log_assert(GetSize(A) == 16);
 
 	SigSpec B = st.sigB;
-	B.extend_u0(16, st.mul->getParam(ID::B_SIGNED).as_bool());
+	B.extend_u0(16, /*st.mul->getParam(ID::B_SIGNED).as_bool()*/ true );
 	log_assert(GetSize(B) == 16);
 
 	SigSpec CD = st.sigCD;
diff --git a/passes/pmgen/ice40_dsp.pmg b/passes/pmgen/ice40_dsp.pmg
index 4de479122..0627961bf 100644
--- a/passes/pmgen/ice40_dsp.pmg
+++ b/passes/pmgen/ice40_dsp.pmg
@@ -303,7 +303,7 @@ reject_ffCD: 		;
 endcode
 
 code sigCD
-	sigCD.extend_u0(32, cd_signed);
+	sigCD.extend_u0(32, /*cd_signed*/ true );
 endcode
 
 code
