diff --git a/Makefile b/Makefile
index e6a27de0..31f9c3b0 100644
--- a/Makefile
+++ b/Makefile
@@ -122,8 +122,8 @@ export PATH := $(PORT_PREFIX)/bin:$(PATH)
 endif
 
 else
-LDFLAGS += -rdynamic
-LDLIBS += -lrt
+# LDFLAGS += -rdynamic
+# LDLIBS += -lrt
 endif
 
 YOSYS_VER := 0.9+4008
@@ -139,7 +139,7 @@ bumpversion:
 # is just a symlink to your actual ABC working directory, as 'make mrproper'
 # will remove the 'abc' directory and you do not want to accidentally
 # delete your work on ABC..
-ABCREV = 4f5f73d
+ABCREV = default
 ABCPULL = 1
 ABCURL ?= https://github.com/YosysHQ/abc
 ABCMKARGS = CC="$(CXX)" CXX="$(CXX)" ABC_USE_LIBSTDCXX=1
