diff --git a/Makefile b/Makefile
index ec8b38c4..d6f89548 100644
--- a/Makefile
+++ b/Makefile
@@ -13,7 +13,7 @@ CONFIG := clang
 ENABLE_TCL := 1
 ENABLE_ABC := 1
 ENABLE_GLOB := 1
-ENABLE_PLUGINS := 1
+ENABLE_PLUGINS := 0
 ENABLE_READLINE := 1
 ENABLE_EDITLINE := 0
 ENABLE_VERIFIC := 0
@@ -119,8 +119,8 @@ export PATH := $(PORT_PREFIX)/bin:$(PATH)
 endif
 
 else
-LDFLAGS += -rdynamic
-LDLIBS += -lrt
+# LDFLAGS += -rdynamic
+# LDLIBS += -lrt
 endif
 
 YOSYS_VER := 0.9+2406
@@ -136,7 +136,7 @@ bumpversion:
 # is just a symlink to your actual ABC working directory, as 'make mrproper'
 # will remove the 'abc' directory and you do not want to accidentally
 # delete your work on ABC..
-ABCREV = 341db25
+ABCREV = default
 ABCPULL = 1
 ABCURL ?= https://github.com/YosysHQ/abc
 ABCMKARGS = CC="$(CXX)" CXX="$(CXX)" ABC_USE_LIBSTDCXX=1
